### ISSUE TITLE
Add Be The Change (BTC) model.

### ISSUE THREAD

**Issue by blaz-r** [microsoft/torchgeo#2918]
### Summary

I'm the author of Be The Change (BTC) model designed for bi-temporal remote sensing change detection.
@isaaccorley suggested on Linkedin that the model could be implemented in TorchGeo. 
If you find this suitable. I now have some time to implement it, but would have some questions.

### Rationale

The main point of BTC is that it's a relatively simple method that achives great results as a result of good selection of core components (like backbone pretraining, loss, correct augmentations...). I believe it can also serve as a solid starting point for future methods.

### Implementation

To add BTC to torchgeo, I belive that majority of components are already supported, but I would need to add:
 - [ ] port hugginface swin weights to torchvision (already have that working) and integrate loading into TorchGeo
 - [ ] add dice loss
 - [ ] modify / somehow change the default augmentations of CD datasets. 

What would be the best way to approach this? It could all be part of a single PR, or each its own with a final BTC PR that merges them all.

### Additional information

Paper arxiv: https://arxiv.org/abs/2507.03367
Original repo: https://github.com/blaz-r/BTC-change-detection

**Comment by adamjstewart** [comment]
Yes, we would love to have your model in TorchGeo! Some comments:

* The majority of models in TorchGeo are based on timm, not torchvision. If your model is already compatible with timm's Swin ViT implementation, it should be very easy to add the model to https://github.com/microsoft/torchgeo/blob/main/torchgeo/models/swin.py.
* Is Dice part of the model architecture, or just used to train the model? We don't necessarily need to add it if it isn't part of the model architecture, and it may make more sense to contribute directly to PyTorch.
* Same here, just adding the pretrained model weights does not require all datasets to be made compatible. What kind of augmentations are you proposing? They need to remain compatible with all of our other CD models too.

**Comment by isaaccorley** [comment]
- we only chose torchvision swin because the Satlas models use this implementation. Preferably we could use timm however if you already have them working with the torchvision swin then that is also fine
- smp actually has https://smp.readthedocs.io/en/latest/losses.html#diceloss implementation so likely don't need to add this unless yours is customized
- regarding the augmentations/transforms, we do allow for these to be added along with the models but detached from the datasets/datamodules so the user can choose to use them or not. I think it makes sense to add them into the same script containing the model. Note that we use kornia so preferably it would be compatible here so that we can leverage kornia's ability to augment both image/mask/bbox appropriately at the batch level.

**Comment by blaz-r** [comment]
Thanks for the replies. To make things clearer on why I want to "add" dice and augmentations. In the paper, we've found that besides the pertaining, simple flip augmentations significantly boost change detection performance, the dice loss also does a bit in some cases. That is why I wanted to introduce this here, and both should really be compatible with any CD method.

By "adding the dice loss" I meant to add it to change the trainer here:
https://github.com/microsoft/torchgeo/blob/834a61ee9648ad36a41bdc88f65445b7dfc69409/torchgeo/trainers/change.py#L94

And about the weights: they are based on the HuggingFace CityScapes segmentation pretrained Swin, so I had to change some things (rename and stack qkv matrices) to make them compatible with torchvision. Segmentation pretrained weights (although not on the RS domain) are quite useful for the CD task, so I wanted to have that option. But as far as I see, timm doesn't have any segmentation pretrained weights.

And about augmentations, the most effective turned out to be a simple rotation + flipping augmentation. This is, of course, suitable for all other models. I know that augmentations are usually not tightly coupled with the model, but given their large influence on performance, I was wondering if there is any way to do that. I will use Kornia, but was just wondering what's the best way to go about this "augmentations & model" coupling. Maybe adding a script to `.experiments`?

**Comment by adamjstewart** [comment]
> In the paper, we've found that besides the pertaining, simple flip augmentations significantly boost change detection performance, the dice loss also does a bit in some cases. That is why I wanted to introduce this here, and both should really be compatible with any CD method.

This is likely true for all remote sensing applications (classification, regression, semantic segmentation, etc.), not just CD. I agree, we should probably change the augmentation defaults and add dice loss for all datamodules/trainers. But that can be a separate PR. This isn't really related to the BTC model itself.

> But as far as I see, timm doesn't have any segmentation pretrained weights.

Timm is only for encoders, we use SMP for semantic segmentation decoders and torchvision for object detection/instance segmentation decoders. We can definitely add your pretrained weights for both encoder and decoder. See what @isaaccorley did in https://github.com/microsoft/torchgeo/blob/main/torchgeo/models/unet.py for an example of how to do this.

> I know that augmentations are usually not tightly coupled with the model, but given their large influence on performance, I was wondering if there is any way to do that. I will use Kornia, but was just wondering what's the best way to go about this "augmentations & model" coupling. Maybe adding a script to .experiments?

We are trying to deprecate the `experiments` directory. The goal is to make the library as flexible as possible. So the defaults should be useful, but if someone wants to change the defaults, we don't want to prevent that. The best you can do for now to couple augmentations and models is to set the `transform` attribute of the weights and hope that users actually use it.

**Comment by blaz-r** [comment]
Thanks for the replies. I'll then start by adding dice to the change module and see how that'll go. I'll keep the extension to other trainers in mind.
Thanks for the link for unet weights example, I'll stick to this style then.
And finally, thanks for the tip on augmentations. I'll add the transforms that we found work well, and we'll then hope the users use them 😄 
I'll get this implemented when I get some time, and then open a PR when I have things working. Thanks again for all the tips.

**Comment by adamjstewart** [comment]
Closed by #2926
